### ISSUE TITLE
Simplecov support

### ISSUE THREAD

**Issue by priverop** [willbryant/parallel_rspec#9]
Hello! I've been using this Gem for a while and it works like a charm, good job!

However, I'm trying to use the [SimpleCov gem](https://github.com/simplecov-ruby/simplecov) to get the Coverage of the project but it doesn't work correctly:

<img width="834" alt="image" src="https://github.com/willbryant/parallel_rspec/assets/1587633/0b056a93-169b-4b59-8479-88671291d291">

It looks like it's only getting the results of one of the workers. 

Has anybody experienced something similar? What do you use to get the test coverage of your project with this tool? 

Thanks!

**Comment by willbryant** [comment]
Yes, that's a known issue. Last time I looked (years ago) there was no parallelism support in SimpleCov, so it's great that they have something now. We could set those vars and see what happens? From a quick skim of the code it looks like maybe SimpleCov is trying to drive the parallelism though? eg. it loads `parallel_tests` in `make_parallel_tests_available`.

**Comment by priverop** [comment]
Thank you for answering so quickly!!

You are right, apparently they require parallel_tests if you have these environment vars... I didn't see that before, maybe it's not as easy as I thought.

I'll create an issue to see their opinion on this.

**Comment by mscottford** [comment]
I think this could be resolved with some documentation changes. Here's the configuration that's working for me. I've added _way_ more comments than I would normally leave in to explain what's being done in detail.


```ruby
# spec_helper.rb

require 'parallel_rspec'
require 'simplecov'

def start_coverage
  # Use the same configuration settings for both the parent process and the forked processes
  SimpleCov.start 'rails' do
    enable_coverage :branch
    primary_coverage :branch
  end
end

# The parent process should show the error status
SimpleCov.print_error_status = true

if ParallelRSpec.running?
  # Turn on SimpleCov's support for collecting coverage from forked processes
  SimpleCov.enable_for_subprocesses true

  ParallelRSpec.configure do |config|
    config.after_fork do |worker_number|
      # Use the worker number to build the SimpleCov command name
      SimpleCov.command_name "RSpec Worker #{worker_number}"

      # be quiet, the parent process will be in charge of output and checking coverage totals
      SimpleCov.print_error_status = false

      # The HTML formatter frequently encounters race conditions when run in parallel. 
      # So only use the simple formatter for the forked processes.
      SimpleCov.formatter SimpleCov::Formatter::SimpleFormatter

      # Set the minimum coverage value for the forked process to 0
      SimpleCov.minimum_coverage 0

      # start SimpleCov in the forked process
      start_coverage
    end
  end
end

# Start SimpleCov in the parent process
start_coverage

# Configure the minimum coverage for the main process to use
SimpleCov.minimum_coverage line: 85, branch: 70
```

I cobbled this together by adapting some of the code from the SimpleCov README in a section that talks about configuring [coverage for subprocesses](https://github.com/simplecov-ruby/simplecov?tab=readme-ov-file#running-simplecov-against-subprocesses) that are created with `fork`. The trick was to _not_ use `SimpleCov.at_fork` and instead use ParallelRSpec's `config.after_fork`.
